tests/e2e/scenarios/expose-pods-by-name/test.yml

```diff
@@ -233,7 +252,9 @@
             label_selectors:
               - internal.skupper.io/listener = true
           register: svc_list
-          until: svc_list.resources | length > 2
+          until:
```

Comment:
Where I end up lost for what to do is around here.

So far, the test starts a deployment, waits for pods to be ready, wait for N services to get created (though expose pods by name), and compares pod names with service names. :heavy_check_mark:.

Then the test restarts the deployment and waits for the new replicaset is started. :heavy_check_mark:. Here we wait for the number of services to be what we expect at steady state - which seems to me likely is begging non-deterministic behavior. :x:.

I was thinking we may need something like "service list length is expected AND service list does not contain old pod/service names".

Thoughts?